File: ruby/binarySearch.rb
require 'pry'
array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]

find = 0

def binary_search(arr, find)
  min = 0
  max = arr.length - 1
  while max >= min
    guess = ((max + min) / 2).to_i
    # binding.pry
    if find == arr[guess]
      return guess
    end
    if arr[guess] < find
      min = guess + 1
    else
      max = guess - 1
    end
  end
  -1
end

puts binary_search(array, find)

File: javascript/binarySearch.js
/* Returns either the index of the location in the array,
  or -1 if the array did not contain the targetValue */

const binarySearch = (arr, target) => {
  let min = 0;
  let max = arr.length - 1 ;
  let guess;

  while (max >= min){
    guess = Math.floor((min + max) / 2)
    if (array[guess] === targetValue){ return guess;}
    else if (array[guess] < targetValue){ min = guess + 1; }
    else { max = guess - 1;}
  }
  return -1;
}
